compiler/sem/pragmas.nim

```diff
@@ -160,6 +160,16 @@ proc illegalCustomPragma*(c: PContext; n: PNode, s: PSym): PNode =
     n,
     PAstDiag(kind: adSemIllegalCustomPragma, customPragma: s))
 
+func isLocal(s: PSym): bool =
+  ## Returns whether `s` is a parameter or local `var`/`let`.
+  # XXX: ideally, we could rely on ``sfGlobal``, but for declarations
```

Comment:
instead of "global" are we really after top level module scope, barring forcing global via the pragma?

asking because I can think of a pretty quick PR I can bang out to improve things.

Reply:
I'm not entirely sure. "globals" are a bit fuzzy at the moment, as `var`/`let` definitions placed outside of procedures but *not* at the top-level module scope are also treated as globals (see `semIdentDef`). That is:

```nim
# module scope
block:
  var a = 0 # <- currently treated as a global
```

Aside: I think only definitions in the top-level module scope should be considered "globals". This would significantly simplify the backends, but it would also be a significant language change requiring large adjustments to the test suite.

Reply:
I'm with you on this one, if it's not top level scope it's not global to the module, at least that's where my instincts are at.

Looking at it from the other point of view:

```nim
# module scope
block:
  var a = 0 # should this be global or not
```

Here the location backing `a` the handle is global, while the handle is short lived (scope of the `block`). I guess the question is should we allow for this sort of thing? Peeling back some more layers if we think about a module as a struct + init function, then a global is any field in the struct, while a non-global is a local to the init function body. Going by that, the code in the `block` in the above example seems like something that would occur in the body and we shouldn't lift `a` into the module struct.

I can probably go around in circles for a bit, but I think what's really happening with global is we're decided whether we want to lift it into the struct (location) or not and what level of control do we want to give the user over all that. Layering FFI atop that is where things get a bit messy as the location is valid but depending upon timing the handle isn't and that seems like a bunch of UB lying in wait. 🤔

Hopefully the above isn't all just blindingly obvious. One thing that seems clear is that this might require a fair bit more discussion and any changes are all outside of this PR. 😆